frontend/src/Components/forms/AddUserForm.tsx

```diff
@@ -29,23 +31,29 @@ export default function AddUserForm({
     handleSubmit,
     formState: { errors },
   } = useForm<Inputs>();
+  let users = userData ? userData.data : [];
 
   const onSubmit: SubmitHandler<Inputs> = async (data) => {
+    for (let i = 0; i < users.length; i++) {
```

Comment:
due to how pagination is handled on the back-end, we might need to add maybe a query string arg, on the show-user endpoint that returns whether it's a valid/available username?

Reply:
I see what you mean, I didn't even consider pagination.  Glad you caught that.

Reply:
Moved logic to test for existing usernames to the backend.
There was an issue with getUserByUsername in users.go, where an empty struct was being returned when no match instead of nil, where nil is the expected no match value throughout other code in this project (ory.go, create user test), so I fixed that as well as added a getUserByEmail function.
Should be all set.

Reply:
Ok cool good catch there. i honestly thought i removed all instances of that method in exchange for one that returned an error but i guess i didn't merge it.